### ISSUE TITLE
jsx-equals-spacing should have an option for ignoring newlines

### ISSUE THREAD

**Issue by LordSputnik** [yannickcr/eslint-plugin-react#527]
In one of my React components, I have the following code snippet:

```
                            <p dangerouslySetInnerHTML=
                                {this.renderedRelationship()}
                            />
```

The `{this.renderedRelationship()}` is split onto a new line, due to a project-standard line length limit of 80 characters. However, jsx-equals-spacing flags this as an issue, so there is a conflict with the max-len rule here.

I'd like to suggest an option to allow newlines (and preceeding whitespace on the next line) to be ignored for this rule. Eg.

```
{
    "ignoreNewlines": true
}
```

I'm not sure what generally happens with always/never when options are added - perhaps that could become a "spacing" option?

**Comment by ljharb** [comment]
Does:
```js
<p
  dangerouslySetInnerHTML={this.renderedRelationship()}
/>
```
not work with your projects styles? Alternatively, put `this.renderedRelationship()` in an intermediate variable? Line breaks after `=` are pretty atypical in JS.
